### ISSUE TITLE
Improve page speed scores

### ISSUE THREAD

**Issue by raen79** [zeit/next.js#11996]
# Feature request

## Is your feature request related to a problem? Please describe.

At the moment the way Next.js loads its CSS and JS doesn't lend itself well to high page speed scores.

## Describe the solution you'd like

This article describes a list of optimizations that can be done to improve pagespeed score using next.js. I think particularly it is important to make the javascript load delay optional as it can cause some applications to break. However those that leverage SSR the best should be able to leverage these optimizations.

https://medium.com/medwing-engineering-product-design/hacking-next-js-for-better-pagespeed-scores-6c651d19f218

## Describe alternatives you've considered

Traditional optimizations were considered but a score nearing 100 is impossible using those.

**Comment by timneutkens** [comment]
> At the moment the way Next.js loads its CSS and JS doesn't lend itself well to high page speed scores.

This is incorrect. The chunking is optimized for achieving interactivity as soon as possible: https://nextjs.org/blog/next-9-2#improved-code-splitting-strategy


Just read through the article and it seems to try and trick lighthouse quite heavily. The comments in the code even refer to that:

```
// 1800ms seems like when PageSpeed Insights stop waiting for more js  
```

What the article achieves is delaying all JS execution for 2 seconds, causing interactivity to break for more than 2 seconds after the page loads, which causes significantly worse user experience as no interactive elements will work.

Basically the TLDR is: Lighthouse can't detect that you're making the overall experience worse as it doesn't have heuristics for this particular edge case that you're using.

**Comment by pjaws** [comment]
I can say that most of my Next.js pages score in the 90s without any "hacking". The ones that don't are mostly due to under-optimized images on my end.

**Comment by johnpolacek** [comment]
@timneutkens Seeing reports of Remove Unused Javascript in a Lighthouse audit of the [vercel.com](https://vercel.com/) homepage. Also reports warnings for  "Minimize main-thread work" and "Reduce JavaScript execution time".  See [Lighthouse report here](https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fvercel.com%2F).

Another example of the Remove Unused Javascript warning is on the @rauchg blog post [that announces Vercel](https://rauchg.com/2020/vercel). See [Lighthouse report here](https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Frauchg.com%2F2020%2Fvercel) which for a minimal page should probably be above 84. By comparison, the [rauchg.com](https://rauchg.com/) homepage still scores 💯 for perf.

I'm in the process of moving my dev blog to Next.js and running into this as well. Seems to be a recent development with Lighthouse making some changes in 6.0 per https://github.com/GoogleChrome/lighthouse/issues/10904
